tests/views/test_player.py

```diff
@@ -0,0 +1,71 @@
+import pytest
+from rest_framework import status
+
+from mars_api.models import Player
+
+pytestmark = pytest.mark.django_db
+
+PLAYER_PATH = "/mars_api/players/"
+
+
+def get_url_with_id(id):
+    return PLAYER_PATH + str(id) + "/"
+
+
+def test_can_get_player_data(api_client, player):
+    """Tests if client can receive player JSON data."""
+    id = player.id
+
+    response = api_client.get(get_url_with_id(id))
+
+    assert response.status_code == status.HTTP_200_OK
+    assert response.data
+    assert response.data["nickname"] == player.nickname
+    assert response.data["motto"] == player.motto
+
+
+def test_create_player_without_user(api_client):
+    """Tests if it's possible to create a player without user."""
+    data = {"nickname": "nicky nick"}
+
+    response = api_client.post(PLAYER_PATH, data)
+    assert response.status_code == status.HTTP_201_CREATED
+    assert len(Player.objects.all()) == 1
+
+    p = Player.objects.get(nickname=data["nickname"])
+    assert isinstance(p, Player)
+    assert p.nickname == data["nickname"]
+
+
+def test_delete_player(api_client, player):
+    """Tests if a player can be deleted provided their nickname."""
+    response = api_client.delete(get_url_with_id(player.id))
+
+    assert response.status_code == status.HTTP_204_NO_CONTENT
+    with pytest.raises(Player.DoesNotExist):
+        Player.objects.get(nickname=player.nickname)
+
+
+def test_fail_change_to_duplicate_nickname(api_client, player):
+    """Tests if only unique player nicknames are allowed."""
+    player2 = Player.objects.create(nickname="nick2", motto="")
```

Comment:
How do I know what's the player1 nickname? Probs would be best to returned an unsaved `player` instance from the fixture and save it in the tests. This way you can make any modifications before saving if needed and adding to the db is explicit.

Reply:
Uhm, look at the `conftest.py`? it's the same nickname throughout all the tests. Most of the tests require having a `player` already in the database, so that would just unnecessarily "un" _DRY_ the rest of the code.